app/field.py

```diff
@@ -17,20 +17,37 @@ def __init__(self, master, settings):
         self.number_of_bombs.set(settings.number_of_bombs)
         self.bomb_counter = tk.Label(self, textvariable=self.number_of_bombs, font="courier 15 bold", fg="red")
         self.bomb_counter.grid(column=0, columnspan=len(self.columns), row=0)
+        self.folder = os.path.dirname(__file__)
+        self.cell_images = {
+            "closed": tk.PhotoImage(file=(os.path.join(self.folder, "../img/closed.png"))),
+            "last": tk.PhotoImage(file=(os.path.join(self.folder, "../img/boom.png"))),
+            "bomb": tk.PhotoImage(file=(os.path.join(self.folder, "../img/bomb.png"))),
+            "not_bomb": tk.PhotoImage(file=(os.path.join(self.folder, "../img/wrong.png"))),
+            "marked": tk.PhotoImage(file=(os.path.join(self.folder, "../img/marked.png"))),
+            "0": tk.PhotoImage(file=(os.path.join(self.folder, "../img/empty.png"))),
+            "1": tk.PhotoImage(file=(os.path.join(self.folder, "../img/1.png"))),
+            "2": tk.PhotoImage(file=(os.path.join(self.folder, "../img/2.png"))),
+            "3": tk.PhotoImage(file=(os.path.join(self.folder, "../img/3.png"))),
+            "4": tk.PhotoImage(file=(os.path.join(self.folder, "../img/4.png"))),
+            "5": tk.PhotoImage(file=(os.path.join(self.folder, "../img/5.png"))),
+            "6": tk.PhotoImage(file=(os.path.join(self.folder, "../img/6.png"))),
+            "7": tk.PhotoImage(file=(os.path.join(self.folder, "../img/7.png"))),
+            "8": tk.PhotoImage(file=(os.path.join(self.folder, "../img/8.png")))
+        }
         self.grid()
         self.create_field()
 
     # Field creation and configuration
     def create_field(self):
         for i in range(self.rows):
             for j in self.columns:
-                self.cells[j+str(i)] = Cell(self, i, j)
+                self.cells[j+str(i)] = Cell(self, i, j, self.cell_images["closed"], self.cell_images["marked"])
         counter = 0
         # Placing bombs
         while counter < self.number_of_bombs.get():
             cell = self.get_random_cell()
             if not cell.is_bomb():
-                cell.bomb = True
+                cell.add_bomb(self.cell_images["bomb"], self.cell_images["last"])

```

Comment:
Maybe the code could be cleaner if Field did not have to pass its Cell's individual image objects. Perhaps, Cell's could be given access to the whole `cell_images` object and take from it whatever images they might need.

But I think that this refactoring is better done later. Ideally, after there are some automatic tests to prove that the application still works after refactoring.